### PR TITLE
Improve `SinglePhasePoromechanicsEmbeddedFractures` and introduce `SinglePhasePoromechanicsConformingFractures` MGR strategy

### PR DESCRIPTION
**Pull request by victorapm**
- [x] Improves the `SinglePhasePoromechanicsEmbeddedFractures` MGR strategy by using BoomerAMG as an F-relaxation solver to the mechanics block in MGR's second level.
- [x] Introduces an MGR strategy for `SinglePhasePoromechanicsConformingFractures`

### CODE REVIEW

**Comment by CusiniM** [comment]
@victorapm can we now clean this up and merge it? If yes, we should also add the MGR strategy for `SinglePhasePoromechanicsConforming` that @matteofrigo5 has tested.

**Comment by victorapm** [comment]
Yes, please @matteofrigo5 go ahead and add the new strategy in this branch. Thanks!

File: src/coreComponents/linearAlgebra/interfaces/hypre/HypreMGR.cpp [hunk 1]
```diff
@@ -54,10 +55,15 @@ void hypre::mgr::createMGR( LinearSolverParameters const & params,
 
   GEOS_LAI_CHECK_ERROR( HYPRE_MGRCreate( &precond.ptr ) );
 
+  // Set MGR's print level
+  HYPRE_Int logLevel = LvArray::integerConversion< HYPRE_Int >( params.logLevel );
+
+  // Update logLevel (1 is reserved in GEOS to log execution times)
+  logLevel = (logLevel > 0) ? ((logLevel - 1) & ~0x2) : 0;
```
Comment (CusiniM) [inline src/coreComponents/linearAlgebra/interfaces/hypre/HypreMGR.cpp]: is this supposed to be here and added by this PR?

File: src/coreComponents/linearAlgebra/interfaces/hypre/HypreSolver.cpp [hunk 2]
```diff
@@ -77,7 +77,7 @@ void createHypreGMRES( LinearSolverParameters const & params,
   GEOS_LAI_CHECK_ERROR( HYPRE_ParCSRGMRESSetTol( solver.ptr, params.krylov.relTolerance ) );
 
   // Default for now
-  HYPRE_Int logLevel = (params.logLevel >= 3) ? 1 : 0;
+  HYPRE_Int logLevel = (params.logLevel >= 3) ? 2 : 0;
```
Comment (CusiniM) [inline src/coreComponents/linearAlgebra/interfaces/hypre/HypreSolver.cpp:80]: same as the previous comment. This seems a more general change. If we need it I am fine with leaving it in this PR.

File: src/coreComponents/linearAlgebra/interfaces/hypre/mgrStrategies/SinglePhasePoromechanicsConformingFractures.hpp [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+namespace mgr
+{
+
+/**
+ * @brief SinglePhasePoromechanicsConformingFractures strategy.
+ *
+ * dofLabel: 0 = displacement, x-component
+ * dofLabel: 1 = displacement, y-component
+ * dofLabel: 2 = displacement, z-component
+ * dofLabel: 3 = pressure (face-centered lagrange multiplier)
```
Comment (CusiniM) [inline src/coreComponents/linearAlgebra/interfaces/hypre/mgrStrategies/SinglePhasePoromechanicsConformingFractures.hpp]: non è
```suggestion
 * dofLabel: 3 = face-centered lagrange multiplier (all components have the same label) 
```

File: src/coreComponents/linearAlgebra/interfaces/hypre/mgrStrategies/SinglePhasePoromechanicsConformingFractures.hpp [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+{
+
+/**
+ * @brief SinglePhasePoromechanicsConformingFractures strategy.
+ *
+ * dofLabel: 0 = displacement, x-component
+ * dofLabel: 1 = displacement, y-component
+ * dofLabel: 2 = displacement, z-component
+ * dofLabel: 3 = pressure (face-centered lagrange multiplier) 
+ * dofLabel: 4 = pressure (cell elem + fracture elems)
```
Comment (CusiniM) [inline src/coreComponents/linearAlgebra/interfaces/hypre/mgrStrategies/SinglePhasePoromechanicsConformingFractures.hpp]:
```suggestion
 * dofLabel: 4 = pressure (cell elem + fracture elems)
```

File: src/coreComponents/linearAlgebra/interfaces/hypre/mgrStrategies/SinglePhasePoromechanicsConformingFractures.hpp [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+ */
+class SinglePhasePoromechanicsConformingFractures : public MGRStrategyBase< 2 >
+{
+public:
+
+  /**
+   * @brief Constructor.
+   */
+  explicit SinglePhasePoromechanicsConformingFractures( arrayView1d< int const > const & )
+    : MGRStrategyBase( 7 )
```
Comment (CusiniM) [inline src/coreComponents/linearAlgebra/interfaces/hypre/mgrStrategies/SinglePhasePoromechanicsConformingFractures.hpp:62]:
```suggestion
    : MGRStrategyBase( 5 )
```
?

**Review by CusiniM — COMMENTED** [review]
I would check that detail about the labels of the lagrange multipliers

File: src/coreComponents/linearAlgebra/interfaces/hypre/HypreMGR.cpp (hunk 1 above)
Comment (victorapm) [reply src/coreComponents/linearAlgebra/interfaces/hypre/HypreMGR.cpp]: I'll remove it and will add this in the true IMPES PR

File: src/coreComponents/linearAlgebra/interfaces/hypre/HypreSolver.cpp (hunk 2 above)
Comment (victorapm) [reply src/coreComponents/linearAlgebra/interfaces/hypre/HypreSolver.cpp:80]: This one can stay in this PR I guess. It's actually a bug fix that was preventing us to get full convergence history from GMRES

**Review by CusiniM — APPROVED** [review]

File: src/coreComponents/linearAlgebra/interfaces/hypre/HypreMGR.cpp (hunk 1 above)
Comment (CusiniM) [reply src/coreComponents/linearAlgebra/interfaces/hypre/HypreMGR.cpp]: removed it!

File: inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_openingFrac_base.xml [hunk 6]
```diff
@@ -15,7 +15,9 @@
         newtonMaxIter="10"
         maxNumConfigurationAttempts="10"/>
       <LinearSolverParameters
-        directParallel="0"/>
+        solverType="gmres"
```
Comment (CusiniM) [inline inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_openingFrac_base.xml]:
```suggestion
        solverType="direct"
```

File: inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_openingFrac_base.xml [hunk 7]
```diff
@@ -15,7 +15,9 @@
         newtonMaxIter="10"
         maxNumConfigurationAttempts="10"/>
       <LinearSolverParameters
-        directParallel="0"/>
+        solverType="direct"
+        preconditionerType="mgr"
+        logLevel="1024"/>
```
Comment (CusiniM) [inline inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_openingFrac_base.xml]:
```suggestion
```

File: inputFiles/poromechanicsFractures/ExponentialDecayPermeability_conformingFracture_smoke.xml [hunk 8]
```diff
@@ -15,7 +15,7 @@
       zCoords="{ 0, 1 }"
       nx="{ 20 }"
       ny="{ 20 }"
-      nz="{ 1 }"
+      nz="{ 2 }"
```
Comment (CusiniM) [inline inputFiles/poromechanicsFractures/ExponentialDecayPermeability_conformingFracture_smoke.xml]:
```suggestion
      nz="{ 1 }"
```

File: inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_openingFrac_horizontal_smoke.xml [hunk 9]
```diff
@@ -15,7 +15,7 @@
       zCoords="{ 0, 1 }"
       nx="{ 4 }"
       ny="{ 2 }"
-      nz="{ 1 }"
+      nz="{ 2 }"
```
Comment (CusiniM) [inline inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_openingFrac_horizontal_smoke.xml]:
```suggestion
      nz="{ 1 }"
```

File: inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_openingFrac_vertical_benchmark.xml [hunk 10]
```diff
@@ -15,7 +15,7 @@
       zCoords="{ 0, 1 }"
       nx="{ 100 }"
       ny="{ 100 }"
-      nz="{ 1 }"
+      nz="{ 2 }"
```
Comment (CusiniM) [inline inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_openingFrac_vertical_benchmark.xml]:
```suggestion
      nz="{ 1 }"
```

File: inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_openingFrac_vertical_smoke.xml [hunk 11]
```diff
@@ -15,7 +15,7 @@
       zCoords="{ 0, 1 }"
       nx="{ 8 }"
       ny="{ 10 }"
-      nz="{ 1 }"
+      nz="{ 2 }"
```
Comment (CusiniM) [inline inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_openingFrac_vertical_smoke.xml]:
```suggestion
      nz="{ 1 }"
```

File: inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_openingFrac_vertical_smoke.xml [hunk 12]
```diff
@@ -128,7 +128,7 @@
       fieldName="totalDisplacement"
       component="2"
       scale="0.0"
-      setNames="{ bottomFace }"/>
+      setNames="{ zneg, zpos }"/>
```
Comment (CusiniM) [inline inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_openingFrac_vertical_smoke.xml]:
```suggestion
      setNames="{bottomFace }"/>
```

File: inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_openingFrac_horizontal_smoke.xml [hunk 13]
```diff
@@ -125,7 +125,7 @@
       fieldName="totalDisplacement"
       component="2"
       scale="0.0"
-      setNames="{ bottomFace }"/>
+      setNames="{ zneg, zpos }"/>
```
Comment (CusiniM) [inline inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_openingFrac_horizontal_smoke.xml]:
```suggestion
      setNames="{ bottomFace }"/>
```

File: inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_openingFrac_vertical_smoke.xml [hunk 14]
```diff
@@ -128,7 +128,7 @@
       fieldName="totalDisplacement"
       component="2"
       scale="0.0"
-      setNames="{ bottomFace }"/>
+      setNames="{bottomFace }"/>
```
Comment (CusiniM) [inline inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_openingFrac_vertical_smoke.xml]:
```suggestion
      setNames="{ bottomFace }"/>
```

**Pull request merged**